src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1467,26 +1501,59 @@ bool OutputStateMachineEngine::_ParseHyperlink(const std::wstring_view string,
 
 // Routine Description:
 // - OSC 10, 11, 12 ; spec ST
-//      spec: The colors are specified by name or RGB specification as per XParseColor
+//      spec: The colors are specified by name or RGB specification as per
+//
+//   It's possible to have multiple "spec", which by design equals to a series of OSC command
+//   with accumulated Ps. For example "OSC 10;color1;11;color2" is effectively an "OSC 10;color1"

```

Comment:
That's not how OSC 10..17 chaining works. It's `OSC 10;color1;color2`, there's no `11;` in the middle, it's implicit that each colour after the first is for the next entry after the starting point.

I think the code is correct, it just looks like the comment is wrong?

Reply:
Yes. I understand it correctly. Guess I’m just bad at writing comments.😅